ninja/signature/details.py

```diff
@@ -222,9 +222,12 @@ def _get_param_type(self, name: str, arg: inspect.Parameter) -> FuncParam:
 
         if get_origin(annotation) is Annotated:
             args = get_args(annotation)
-            if isinstance(args[1], Param):
+            if isinstance(args[-1], Param):
                 prev_default = default
-                annotation, default = args
+                if len(args) == 2:
+                    annotation, default = args
+                else:
+                    annotation, default = Annotated[*args[:-1]], args[-1]
```

Comment:
It looks like this runtime re-wrapping produces a syntax error in some versions of Python. I'm not sure what a good alternative would be but am open to suggestions, @vitalik